apps/remix-ide/src/app/editor/SourceHighlighters.js

```diff
@@ -28,13 +25,40 @@ class SourceHighlighters {
     }
   }
 
+  highlightAllFrom (from) {
+    try {
+      if (!this.highlighters[from]) return
+      const sourceHighlight = new SourceHighlighter()
+      for (const index in this.highlighters[from]) {
+        sourceHighlight.currentSourceLocationFromfileName(
+          this.highlighters[from][index].position,
+          this.highlighters[from][index].filePath,
```

Comment:
in which case do you have `filePath` here ? I didn't see `filePath` but `source` is present. Also `hexColor` seems not

Reply:
^ might be the reason why you have `filePath` undefined and that you needed to to `if (!this.source) this.source = this._deps.fileManager.currentFile()`